File: problems/Majority Element/majorityElement.cs
public class Solution {
    // Boyer-Moore Voting Algorithm
    public int MajorityElement(int[] nums) {
        var count = 0;
        var candidate = -1;

        foreach (var num in nums) {
            if (count == 0) {
                candidate = num;
            }
            count += (num == candidate) ? 1 : -1;
        }

        return candidate;
    }
}

File: problems/Majority Element/majorityElement.java
class Solution {
    public int majorityElement(int[] nums) {
        int count = 0;
        int candidate = 0;

        for (int num : nums) {
            if (0 == count) {
                candidate = num;
            }
            count += candidate == num ? 1 : -1;
        }

        return candidate;
    }
}
